infra/RepoManager.py

```diff
@@ -0,0 +1,220 @@
+  """Class to manage git repos from python.
+
+  Attributes:
+    repo_url: The location of the git repo
+    local_dir: The location of where the repo clone is stored locally
+    repo_name: The name of the github project
+    repo_dir: The location of the main repo
+    full_path: The full filepath location of the main repo
+  """
+  repo_url = ''
+  repo_name = ''
+  repo_dir = ''
+  local_dir = ''
+  full_path = ''
+
+  def __init__(self, repo_url, commit=None, local_dir='tmp'):
+    """Constructs a repo manager class.
+
+    Args:
+      repo_url: The github url needed to clone
+      commit: The specified commit to be checked out
+      local_dir: The local location the repo will live in
+    """
+
+    self.repo_url = repo_url
+    self.local_dir = local_dir
+    self.repo_name = self.repo_url.split('/')[-1].strip('.git')
+    self.repo_dir = os.path.join(self.local_dir, self.repo_name)
+    self.full_path = os.path.join(os.getcwd(), self.repo_dir)
+    self._clone()
+    if branch is not None:
+      self.checkout(branch)
+
+    if commit is not None:
+      self.checkout_commit(commit)
+
+  def _clone(self):
+    """Creates a clone of the repo in the specified directory.
+
+      Raises:
+        RepoManagerException if the repo was not able to be cloned
+    """
+    if not os.path.exists(self.local_dir):
+      os.makedirs(self.local_dir)
+    self.remove_repo()
+    _, err = self._run_command(['git', 'clone', self.repo_url], self.local_dir)
+    if err is not None:
+      raise RepoManagerException(
+          'Failed cloning repo %s, with error %s)' % (self.repo_url, err))
+    if not self._is_git_repo():
+      raise RepoManagerException('%s is not a git repo' % self.repo_url)
+
+  def _run_command(self, command, location='.'):
+    """ Runs a shell command in the specified directory location.
+
+    Args:
+      command: The command as a list to be run
+
+    Returns:
+      The stdout of the command, the stderr of the command
+    """
+    cur_dir = os.getcwd()
+    os.chdir(location)
+    process = subprocess.Popen(command, stdout=subprocess.PIPE)
+    out, err = process.communicate()
+    os.chdir(cur_dir)
+    if err is not None:
+      err = err.decode('ascii')
+      print('Error %s running command %s' % (err, command))
+    if out is not None:
+      out = out.decode('ascii')
+    return out, err
+
+  def _is_git_repo(self):
+    """Test if the current repo dir is a git repo or not.
+
+    Returns:
+      True if the current repo_dir is a valid git repo
+    """
+    git_path = os.path.join(self.repo_dir, '.git')
+    return os.path.isdir(git_path)
+
+  def _commit_exists(self, commit):
+    """ Checks to see if a commit exists in the project repo.
+
+    Args:
+      commit: The commit SHA you are checking for
+      project_name: The name of the project you are checking
+
+    Returns:
+      True if the commit exits in the project
+    """
+
+    # Handle the default case
+    if commit.strip(' ') == '':
```

Comment:
no need for the explicit ' ' ? We can just strip all whitespace (which is the default behaviour of .strip() with no args)